examples/10_streaming_write.py

```diff
@@ -3,31 +3,92 @@
 import numpy as np
 import sys
 
-if __name__ == "__main__":
-    if 'adios2' not in io.variants or not io.variants['adios2']:
-        print('This example requires ADIOS2')
-        sys.exit(0)
-
-    series = io.Series("stream.sst", io.Access_Type.create)
+# pass-through for ADIOS2 engine parameters
+# https://adios2.readthedocs.io/en/latest/engines/engines.html
+config = """
+{
+  "adios2": {
+    "engine": {
+      "parameters": {
+        "Threads": "4"
+      }
+    },
+    "dataset": {
+      "operators": [
+        {
+          "type": "bzip2"
+        }
+      ]
+    }
+  }
+}
+"""
 
+if __name__ == "__main__":
+    # this block is for our CI, SST engine is not present on all systems
     backends = io.file_extensions
     if "sst" not in backends:
         print("SST engine not available in ADIOS2.")
         sys.exit(0)
 
-    datatype = np.dtype("double")
-    length = 10
-    global_extent = [10]
-    dataset = io.Dataset(datatype, global_extent)
+    # create a series and specify some global metadata
+    # change the file extension to .json, .h5 or .bp for regular file writing
+    series = io.Series("simData.sst", io.Access_Type.create, config)
+    series.set_author("Franz")
+    series.set_software("python snippet")
```

Comment:
Maybe more concrete:
```suggestion
    series.set_software("openPMD-api-python-examples")
```